src/jdk.naming.dns/share/classes/com/sun/jndi/dns/DnsContext.java

```diff
@@ -176,13 +176,13 @@ public Object addToEnvironment(String propName, Object propVal)
         } else if (propName.equals(INIT_TIMEOUT)) {
             int val = Integer.parseInt((String) propVal);
             if (timeout != val) {
-                timeout = val;
+                timeout = Math.max(val, 0);
                 resolver = null;
             }
         } else if (propName.equals(RETRIES)) {
             int val = Integer.parseInt((String) propVal);
             if (retries != val) {
-                retries = val;
+                retries = Math.clamp(val, 1, 31);
```

Comment:
Maybe we should rather use 30 here, since `1 << 1` is 2 and `1 << 31` is negative